src/radio.rs

```diff
@@ -8,66 +8,163 @@ use std::{
+            // Receive a Okay response
+            let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?; //[0] [1]
+        };
 
         Ok(true)
     }
 
-    /// Receive information from a TcpStream 
-    fn receive_information(tcp_connection: &TcpStream) -> Result<Vec<R>, Box<dyn Error>> {
-        let tcp_stream_ref: Rc<TcpStream> = Rc::new(tcp_connection.try_clone()?);
+    fn receive_chucked(tcp_connection: &TcpStream) -> Result<Vec<u8>, Box<dyn Error>> {
+        let tcp_connection_ref: Rc<TcpStream> = Rc::new(tcp_connection.try_clone()?);
+
+        // Size of the information what will come
+        let size: usize =
+            bincode::deserialize(&Self::receive_bytes(128, tcp_connection_ref.try_clone()?)?)?;
+        println!("SIZE: {:#?}", size);
 
-        let information_size: usize =
-            bincode::deserialize(&Self::recv_data(512, tcp_stream_ref.try_clone()?)?)?;
+        let chucks = size/128;
+        println!("CHUCKS: {:#?}", chucks);
+        let last_chuck_size = size % 128;
+        println!("LAST CHUCK SIZE: {:#?}", last_chuck_size);
 
-        Self::send_data(&[1], tcp_stream_ref.try_clone()?)?;
+        // Send the chunks
+        Self::send_bytes(&bincode::serialize(&chucks)?, tcp_connection_ref.try_clone()?)?;
+        // Receive a Okay response
+        let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?; //[0] [1]
 
-        let commands = bincode::deserialize::<Vec<R>>(&Self::recv_data(
-            information_size,
-            tcp_stream_ref.try_clone()?,
-        )?)?;
+        // Send the last chunk size
+        Self::send_bytes(&bincode::serialize(&last_chuck_size)?, tcp_connection_ref.try_clone()?)?;
 
-        Ok(commands)
+        // Receive a Okay response
+        let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?; //[0] [1]
+
+        // Send a Okay response
+        Self::send_bytes(&[1], tcp_connection_ref.try_clone()?)?;
+
+        let mut data: Vec<u8> = vec![];
+        println!("{:?}", data);
+        for i in 0..=chucks+1 {
+            if i+1 == chucks+1 {
+                let mut data_received= Self::receive_bytes(last_chuck_size, tcp_connection_ref.try_clone()?)?;
+                println!("RECEIVED DATA CHUCK: {:?}", data_received);
+                data.append(&mut data_received);
+                // Send a Okay response
+                Self::send_bytes(&[1], tcp_connection_ref.try_clone()?)?;
+                break;
+            }
+            // Receive data
+            let mut data_received= Self::receive_bytes(128, tcp_connection_ref.try_clone()?)?;
+            println!("RECEIVED DATA CHUCK: {:?}", data_received);
+            data.append(&mut data_received);
+            // Send a Okay response
+            Self::send_bytes(&[1], tcp_connection_ref.try_clone()?)?;
+        };
+
+        println!("RECEIVED DATA: {:?}", data);
```

Comment:
But when the other get the bytes, the total data differs only per 8 bytes in front of the original sent data.